wait/wait_test.go

```diff
@@ -91,8 +86,6 @@ func TestWaitableContext_Concurrency(t *testing.T) {
 	// a map to check the status of goroutines
 	goroutines := sync.Map{}
 
-	// a waitable context
-	ctx, wait := WithWait(ctx)
```

Comment:
I think we should keep this and instead remove the declaration below, so that when we `wait()` at the end of the test, we're waiting for the parent context of the two child contexts marked as `Done(ctx)` in the goroutines.